gremlin/agent_apparmor.profile

```diff
@@ -0,0 +1,107 @@
+#include <tunables/global>
+
+
+profile gremlin-agent flags=(attach_disconnected,mediate_deleted) {
+  #include <abstractions/base>
+
+  network,
+  file,
+  umount,
+  mount,
+
+  /entrypoint.sh rix,
+  /var/lib/gremlin/** rwix,
+  /var/log/gremlin/** rwix,
+  /etc/gremlin/** rwix,
+
+  # /dev/null
+  /dev rix,
+  /dev/null rwlix,
+
+  # Container runtime
+  /run/docker/runtime-runc/moby rwix,
+  /var/run/docker.sock rwix,
+  /run/crio/crio.sock rwix,
+  /run/runc rwix,
+  /run/containers/containers.sock rwix,
+  /run/containerd/runc/k8s.io rwix,
+
+  # We need access to Pid 1's real pid to resolve the container driver
+  # we're just taking a read perm here to accomlish this
+  file r @{PROC}/1/ns/pid,
+  ptrace read,
+
+
+  # Attack capabilities
+  /proc/sysrq-trigger w,
+  /sys/fs/cgroup/** rw,
+  /proc/** rl,
+  # In order to join target container network space
+  @{PROC}/[0-9]+/ns/net w,
+  # In order to assume the root of the target container
+  pivot_root,
+
+  # Needed for specific Gremlin Attacks
+  capability sys_boot,
+  capability sys_time,
+  capability sys_admin, # needed for setns
+  capability net_admin,
+  capability kill,
+  capability setfcap,
+  capability audit_write,
+  capability mknod,
+
+  # Needed to execute attacks
+  capability net_bind_service,
+  capability setuid,
+  capability setgid,
+  capability chown,
```

Comment:
> What was your process to find out the list of all possible capabilities needed for this AppArmor profile

Manual testing.  I tried removing each of these individually.

It's worth noting that https://www.gremlin.com/docs/security/overview/ is written based on assumptions about the installation environment.  It's possible to create an installation environment (and people commonly do) where we don't need `setuid` and friends, but that's not true in kubernetes installations that are locked down with apparmor which I tested

It might be worth taking a pass over https://www.gremlin.com/docs/security/overview/ again? that's sort of beyond the scope of this change